app.js

```diff
@@ -56,19 +49,17 @@ async function init() {
 
 async function assignNextUser(issueID) {
   while (app.allUsers && app.allUsers.length > 0) {
-    // Reset queue if empty by copying from master list
-    if (app.queuedUsers && app.queuedUsers.length === 0) {
-      repopulateUserQueue();
-    }
-    const dequeuedUser = app.queuedUsers.shift();
+    let nextUser = app.allUsers[app.nextUserIndex];
     try { 
-      let result = await assignIssue(issueID, dequeuedUser);
+      let result = await assignIssue(issueID, nextUser);
+      // Advance userIndex to next user (wrap around based on length of array)
+      app.nextUserIndex = (++app.nextUserIndex) % app.allUsers.length;
```

Comment:
@manuzope thanks for your notes :) I updated everything to use an index to cycle through users and updated the tests accordingly in this commit... but for this part, uh is this too hacky? Suggestions?

Reply:
Nope, incrementing the index like this is great!

Reply:
So, one of the problems of using an index, is that `nextUserIndex` will be the same between requests. Remember: while we're waiting for `assignIssue` to resolve (asynchronous request), Express will happily accept another webhook request and try to `assignIssue` to another user -- in this case the _same_ user because `nextUserIndex` doesn't increment until after `assignIssue` returns.